torch/fft/__init__.py

```diff
@@ -9,18 +9,227 @@
 # connects the torch.fft Python namespace to the torch._C._fft builtins.
 
 fft = _add_docstr(_fft.fft_fft, r"""
-fft(input) -> Tensor
+fft(input: Tensor, n: Optional[int] =None, dim: int =-1, norm: Optional[str] =None) -> Tensor
 
 Computes the one dimensional discrete Fourier transform of :attr:`input`.
 
 Args:
   input (Tensor): the input tensor
+  n (int, optional): FFT length. If given, the input will either be zero-padded
+    or trimmed to this length before taking the FFT.
+  dim (int, optional): The dimension along which to take the one dimensional FFT.
+  norm (str, optional): Normalization mode:
```

Comment:
In NumPy norm is None or "ortho." Having a superset (or subset) of NumPy functionality is OK, but I'm confused by the "forward" and "backward" here since "backward" says it's normalizing `torch.fft.ifft`. Can this clarify what's going on?

Also, the default from the signature is None, not "backward." This could say "if norm is None then 'backward' normalization is used?"

Reply:
These will be in the next NumPy and SciPy releases. See numpy/numpy#16476 and scipy/scipy#12532.

Reply:
And `None` is the same as `"backward"` which is why that is given as the default. This is consistent with the numpy docstring.

Reply:
That's cool. Maybe it's just confusing to me? I still think we could improve the clarity of what's happening here for readers.

Reply:
Reworded to "the {forward,backward} transform ({fft,ifft}) is normalized by `1/n`. Hopefully that is clearer.